utils/reconciler.go

```diff
@@ -383,61 +447,13 @@ func (r *ReconcilerBase) GenerateSvcCertSecret(ba common.BaseComponent, prefix s
 	} else if ok {
 		bao := ba.(metav1.Object)
 
-		issuer := &certmanagerv1.Issuer{ObjectMeta: metav1.ObjectMeta{
-			Name:      prefix + "-self-signed",
-			Namespace: bao.GetNamespace(),
-		}}
-		err = r.CreateOrUpdate(issuer, nil, func() error {
-			issuer.Spec.SelfSigned = &certmanagerv1.SelfSignedIssuer{}
-			issuer.Labels = MergeMaps(issuer.Labels, map[string]string{"app.kubernetes.io/managed-by": operatorName})
-			return nil
-		})
-		if err != nil {
-			return true, err
-		}
-		caCert := &certmanagerv1.Certificate{ObjectMeta: metav1.ObjectMeta{
-			Name:      prefix + "-ca-cert",
-			Namespace: bao.GetNamespace(),
-		}}
-		err = r.CreateOrUpdate(caCert, nil, func() error {
-			caCert.Labels = MergeMaps(caCert.Labels, map[string]string{"app.kubernetes.io/managed-by": operatorName})
-			caCert.Spec.CommonName = CACommonName
-			caCert.Spec.IsCA = true
-			caCert.Spec.SecretName = prefix + "-ca-tls"
-			caCert.Spec.IssuerRef = certmanagermetav1.ObjectReference{
-				Name: prefix + "-self-signed",
-			}
-
-			duration, err := time.ParseDuration(common.Config[common.OpConfigCMCADuration])
-			if err != nil {
-				return err
+		err = r.GenerateCMIssuer(bao.GetNamespace(), prefix, CACommonName, operatorName)
+		if err !=nil {
+			if errors.Is(err, APIVersionNotFoundError) {
+				return false, nil
```

Comment:
Should this be `return false, err` ? similar to how _Certificate_ is handled ([here](https://github.com/application-stacks/runtime-component-operator/blob/2227b1491c3d32106d467e78e19dee636d8387d3/utils/reconciler.go#L446))

Reply:
In the original code when certmanager.io API not available it didn't return an error so that reconciler goes on, either using  other method (OCP) or using byo certificate.

Reply:
In the original code when certmanager.io API not available it didn't return an error so that reconciler goes on, either using other method (OCP) or using byo certificate.